### DISCUSSION TITLE
How can I obtain the inertia matrix automatically calculated by Webots when inertiaMatrix is empty?

### DISCUSSION THREAD

**Discussion by DylanYang-Official** [cyberbotics/webots#6522] [General]
Recently, I have been conducting experiments using the DJI Mavic2 PRO robot in Webots. As I need to add an MPC controller, I require its inertia matrix. I noticed that its inertia matrix has not been set, yet it can fly normally. After consulting the manual, I learned that Webots can automatically calculate the inertia matrix of the robot when it is not set. So, how can I obtain the automatically calculated matrix by Webots?
<img width="965" alt="20bdc5b60cf19ec35e2738aaaf4f783" src="https://github.com/cyberbotics/webots/assets/64779837/7679e6b6-8d5a-4761-a0d8-de296647577e">
![20240318183111](https://github.com/cyberbotics/webots/assets/64779837/4882acc4-3487-4ee6-94d6-a8bd00cdab00)

**Comment by ShuffleWire** [comment]
Maybe this can partially help : 

You can get the inertia matrix of the body of the drone by clicking on it's main node Robot "Mavic 2 PRO" and then in the Mass tab you can find the inertia matrix "Excluding descendant" (which is most of the mass I suppose)

![image](https://github.com/cyberbotics/webots/assets/83014015/bfa181fa-219b-4655-9082-a40ccb73bcc7)

**Reply by DylanYang-Official** [reply]
THANK YOU !!!!!
